lib/paramedic.js

```diff
@@ -106,14 +110,21 @@ ParamedicRunner.prototype = {
         }).then(function(res) {
             if (self.testsPassed) {
                 logger.info("All tests have been passed.");
-                process.exit(0);
+                return TESTS_PASS;
             } else {
                 logger.error("There are tests failures.");
-                process.exit(1);
+                return TESTS_FAILURE;
             }
         }, function(err) {
             logger.error("Failed: " + err);
-            process.exit(2);
+            return NONTESTS_FAILURE;
+        }).then(function(exitCode){
+            if(self.config.isCleanUpAfterRun()) {
+                logger.info("cordova-paramedic: Deleting the application: " + self.tempFolder.name);
+                shell.cd("..");
```

Comment:
Instead of using `shell.cd` here and in `createTempProject`, maybe use `shell.pushd` and `shell.popd` if possible.